packages/cw-core-interface/src/lib.rs

```diff
@@ -1,11 +1,123 @@
-use cosmwasm_std::{CosmosMsg, Empty};
+use cosmwasm_std::{Addr, Binary, CosmosMsg, Empty, WasmMsg};
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
 pub mod voting;
 
+/// The cw-core interface.
 #[derive(Serialize, Deserialize, Clone, Debug, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum ExecuteMsg {
+    /// Causes the core module to execute all of MSGS in order. Only
+    /// callabale by a proposal module.1
     ExecuteProposalHook { msgs: Vec<CosmosMsg<Empty>> },
 }
+
+/// Information about the CosmWasm level admin of a contract. Used in
+/// conjunction with `ModuleInstantiateInfo` to instantiate modules.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum Admin {
```

Comment:
**question**: how come `None` was removed?

Reply:
good q! i feel it's a little more idomatic rust to use an option instead of having our own none variant. this is now specified as `Some(Admin::CoreModule {})` instead of `Admin::CoreModule {}` and the no admin case is now just `None`.